### ISSUE TITLE
undefined method `to_ary` for Enumerator (Rails 7.1.1 & Activeadmin 3.1.0)

### ISSUE THREAD

**Issue by paulhklam1122** [activeadmin/activeadmin#8133]
### Did you find a bug?

The below no method error is thrown when trying to download a CSV from an index page.

```
undefined method `to_ary' for #<Enumerator: #<Enumerator::Generator:0x0000ffff8bc00d60>:each> (NoMethodError)

        @buf.to_ary
            ^^^^^^^
Did you mean?  to_a
```

### Expected behavior

A CSV file should be downloaded with no issues.

### Actual behavior

The above error is thrown by puma.

### How to reproduce

Install Rails 7.1.1
Install Activeadmin 3.1.0

Try to download a CSV from any index page.

[our bug report template]: https://github.com/activeadmin/activeadmin/blob/master/tasks/bug_report_template.rb

**Comment by javierjulio** [comment]
Thank you. Sorry, there isn't anything actionable here. This could depend on several factors. We use Rails 7.1.1 as part of the default dev workflow here and I'm able to export a CSV on any index page using the development app just fine. No reports on my side with our own ActiveAdmin usage either as we run v3.1.0. I'd suggest creating a reproduction to isolate the issue as that would be most helpful for identifying and fixing this. We would be open to a PR to resolve it as well. Thanks.

**Comment by paulhklam1122** [comment]
> Thank you. Sorry, there isn't anything actionable here. This could depend on several factors. We use Rails 7.1.1 as part of the default dev workflow here and I'm able to export a CSV on any index page using the development app just fine. No reports on my side with our own ActiveAdmin usage either as we run v3.1.0. I'd suggest creating a reproduction to isolate the issue as that would be most helpful for identifying and fixing this. We would be open to a PR to resolve it as well. Thanks.

Hi @javierjulio I did some more debugging and it looks like it's due to the Activeadmin CSV streaming option.

If you're testing on development trying adding `config.disable_streaming_in = %w[staging]` to your `config/initializers/active_admin.rb`. This will enable the CSV streaming for development and you will get the previously menntioned error.

**Comment by gsar** [comment]
this is a [regression](https://github.com/rails/rails/issues/49588) in rails 7.1.0. i see there is [a PR](https://github.com/rails/rails/pull/49616) that has been merged into the upcoming 7.1.2 release branch.

**Comment by paulhklam1122** [comment]
Thank you, I also see [another PR ](https://github.com/rails/rails/issues/49588) for the same issue. Thank you all, I guess we'll just have to wait for Rails `7.1.2` to fix this issue.
